near-sdk/src/utils/mod.rs

```diff
@@ -40,6 +40,96 @@ macro_rules! log {
     };
 }
 
+/// Helper macro to create assertions that will panic through the runtime host functions.
+///
+/// This macro can be used similarly to [`assert!`] but will reduce code size by not including
+/// file and rust specific data in the panic message.
+///
+/// # Examples
+///
+/// ```no_run
+/// use near_sdk::require;
+///
+/// # fn main() {
+/// let a = 2;
+/// require!(a > 0);
+/// require!("test" != "other", "Some custom error message if false");
+/// # }
+/// ```
+#[macro_export]
+macro_rules! require {
+    ($cond:expr $(,)?) => {
+        if !$cond {
```

Comment:
cool, seems like these will be less costly on CPU cycles than `assert`s, is that right?

Reply:
I don't think that will necessarily be the case, but it should reduce the code size from not including file/line numbers of where the `assert` is. This also depends on the use case, though.